hydroflow_lang/src/graph/ops/anti_join.rs

```diff
@@ -50,44 +51,110 @@ pub const ANTI_JOIN: OperatorConstraints = OperatorConstraints {
+                    ));
+                    return Err(());
+                }
+            };
+            Ok((antijoindata_ident, borrow_ident, init, borrow))
+        };
+
+        let (pos_antijoindata_ident, pos_borrow_ident, pos_init, pos_borrow) =
+                make_antijoindata(persistences[0], "pos")?;
+        let (neg_antijoindata_ident, neg_borrow_ident, neg_init, neg_borrow) =
+                make_antijoindata(persistences[1], "neg")?;
+
         let write_prologue = quote_spanned! {op_span=>
-            let #handle_ident = #hydroflow.add_state(::std::cell::RefCell::new(
-                #root::util::monotonic_map::MonotonicMap::<_, #root::rustc_hash::FxHashSet<_>>::default(),
+            let #neg_antijoindata_ident = #hydroflow.add_state(std::cell::RefCell::new(
+                #neg_init
+            ));
+            let #pos_antijoindata_ident = #hydroflow.add_state(std::cell::RefCell::new(
+                #pos_init
             ));
         };
-        let write_iterator = {
-            let borrow_ident = wc.make_ident("borrow");
 
-            let input_neg = &inputs[0]; // N before P
-            let input_pos = &inputs[1];
+        let input_neg = &inputs[0]; // N before P
+        let input_pos = &inputs[1];
+        let write_iterator = {
             quote_spanned! {op_span=>
-                let mut #borrow_ident = #context.state_ref(#handle_ident).borrow_mut();
+                let mut #neg_borrow_ident = #context.state_ref(#neg_antijoindata_ident).borrow_mut();
+                let mut #pos_borrow_ident = #context.state_ref(#pos_antijoindata_ident).borrow_mut();    
                 let #ident = {
                     /// Limit error propagation by bounding locally, erasing output iterator type.
                     #[inline(always)]
                     fn check_inputs<'a, K, I1, V, I2>(
-                        input_pos: I1,
-                        input_neg: I2,
-                        borrow_state: &'a mut #root::rustc_hash::FxHashSet<K>,
+                        input_neg: I1,
+                        input_pos: I2,
+                        neg_state: &'a mut #root::rustc_hash::FxHashSet<K>,
+                        pos_state: &'a mut #root::rustc_hash::FxHashSet<(K, V)>,
                     ) -> impl 'a + Iterator<Item = (K, V)>
                     where
                         K: Eq + ::std::hash::Hash + Clone,
-                        V: Eq + Clone,
-                        I1: 'a + Iterator<Item = (K, V)>,
-                        I2: 'a + Iterator<Item = K>,
+                        V: Eq + ::std::hash::Hash + Clone,
+                        I1: 'a + Iterator<Item = K>,
+                        I2: 'a + Iterator<Item = (K, V)>,
                     {
-                        borrow_state.extend(input_neg);
-                        input_pos.filter(move |x| !borrow_state.contains(&x.0))
+                        neg_state.extend(input_neg);
```

Comment:
I think you don't need any of the mapping functionality of `filter_map` and `filter` would be fine to use here.